### PR TITLE
label all Cilium resources with "app.kubernetes.io/part-of: cilium"

### LINKED ISSUE #20088: Label all Cilium resources with a known label for tracking purposes and sysdump
Kubernetes [has](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels) `app.kubernetes.io/part-of: X` to be used for this purpose. We can make use of this with Cilium to make it obvious which resources belong to / used by Cilium. We can also utilize this label to fetch all relevant resources during a sysdump to ensure we don't miss any important information.

Proposal: label resources with `app.kubernetes.io/part-of: cilium`

### PR DESCRIPTION
**Pull request by cyclinder**
Signed-off-by: cyclinder qifeng.guo@daocloud.io

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: [#20088](https://github.com/cilium/cilium/issues/20088)

```release-note
label all Cilium resources with "app.kubernetes.io/part-of: cilium"
```

### CODE REVIEW

**Comment by cyclinder** [comment]
I'm not sure if I need to label resources of the type `configMap`,`RBAC`..

File: install/kubernetes/cilium/templates/cilium-agent/service.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-agent/service.yaml]: Let's not add those to the service selector. I'm only marking the first instance, but there are a few of those.

**Review by gandro — CHANGES_REQUESTED** [review]
Thanks for the PR. Overall I think you got the relevant resources, though I don't think we should also make it part of e.g. the service selectors.

**Comment by ldelossa** [comment]
Is the idea here to eventually deprecate the `k8s-app=cilium` label?

**Comment by gandro** [comment]
> Is the idea here to eventually deprecate the `k8s-app=cilium` label?

I don't think so. `k8s-app` uses different values for each component (e.g. `k8s-app: cilium` vs `k8s-app: hubble-relay`), which we also use for service selectors, while `part-of` is the same for all components of the `cilium` Helm chart and more purely informational.

**Comment by ldelossa** [comment]
Yes but if we begin following the "recommended" labels, `k8s-app` would equate to `app.kubernetes.io/name` or `app.kubernetes.io/component`. See: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

**Comment by ldelossa** [comment]
@gandro and @cyclinder Given that this moves us to one of the recommended k8s labels, I'd also be pretty happy if we moved `k8s-app` to the recommended ones suggested above as well. I do not think we should *remove* the `k8s-app` label just yet, but migrate away from it and towards the set of recommended lables. How do you both feel about this?

**Comment by tommyp1ckles** [comment]
> I'm not sure if I need to label resources of the type `configMap`,`RBAC`..

Is there a reason those might be excluded?

**Comment by tommyp1ckles** [comment]
We use `k8s-app` for all the deployment/ds matchLabels selectors. We would probably want to deprecate that to make that change.

File: install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml
Comment (tommyp1ckles) [inline install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml]: @gandro To your point about service selectors, we probably want to extend that to template selectors?

File: install/kubernetes/cilium/templates/cilium-agent/service.yaml
Comment (cyclinder) [reply install/kubernetes/cilium/templates/cilium-agent/service.yaml]: Thanks your point! @gandro 

Do all `selector` and `matchLabels` not need to be added to _app.kubernetes.io/part-of: cilium_ ?

**Comment by cyclinder** [comment]
> @gandro and @cyclinder Given that this moves us to one of the recommended k8s labels, I'd also be pretty happy if we moved `k8s-app` to the recommended ones suggested above as well. I do not think we should _remove_ the `k8s-app` label just yet, but migrate away from it and towards the set of recommended lables. How do you both feel about this?

Thanks for the review! I totally agree.

We shouldn't remove `k8s-app` right now (maybe we should still keep it), but at least we should tag all cilium resources with `app.kubernetes.io/name `  or  `app.kubernetes.io/component` according to [recommended" labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/), just like `k8s-app`.

**Comment by cyclinder** [comment]
> Is there a reason those might be excluded?

Since I see no labels for these resources at the moment

File: install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml
Comment (cyclinder) [reply install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml]: @tommyp1ckles I think when we labels all cilium resources with `app.kubernetes.io/name` or `app.kubernetes.io/component`, They should also be used for `selector` or `matchLabels`, Just like `k8s-app`, And `app.kubernetes.io/part-of` is only for label.

**Comment by gandro** [comment]
On whether or not to deprecate the `k8s-app` labels: While I'm not against removing them, I believe `cilium sysdump` is currently relying on them. 

So if we decide to remove them, then we need to figure out a migration strategy for `cilium sysdump` that works with both old and new versions of Cilium.

File: install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml
Comment (gandro) [reply install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml]: I gave this some more thought, and yes I actually agree that using both labels (`name` and `component`) in the selectors is better. So please ignore my previous request.

File: install/kubernetes/cilium/templates/cilium-agent/service.yaml
Comment (gandro) [reply install/kubernetes/cilium/templates/cilium-agent/service.yaml]: Sorry about this, as pointed out here https://github.com/cilium/cilium/pull/20213#discussion_r898818078 it seems using both labels is more robust, so I think need to revise my request here.

Are there best practices guidelines for this?

**Comment by ldelossa** [comment]
@gandro and @cyclinder I do agree. We do not want to deprecate `k8s-app` until we can safely do so, especially with its usage in `cilium sysdump`. That being said, we can add the recommended labels like mentioned above. There is a bit of value add here. Right now I have a kind cluster and I see that the 'agent' pods use `k8s-app=cilium` while the operators do not. 

Ideally, we can do something like:
```
app.kubernetes.io/part-of: cilium
app.kubernetes.io/name: cilium-agent
```

```
app.kubernetes.io/part-of: cilium
app.kubernetes.io/name: cilium-operator
```

Tho, I'm not sure if s/name/component or not?

**Comment by cyclinder** [comment]
Update:

- Label all Cilium resources with `app.kubernetes.io/part-of: cilium` (But not including configMap,RBAC.etc, Since these resources do not have any label), The meaning of `app.kubernetes.io/part-of` is just to label which resources belong to cilium, and not be used for `selector` and `matchLabels`.
- The label of  `app.kubernetes.io/name` can be equal to `k8s-app`, which is used to label different components in cilium, and which also used for `selector` and `matchLabels`, Just like `k8s-app` do.  In addition to, I used `app.kubernetes.io/name` instead of `name` .

@ldelossa @gandro

File: install/kubernetes/cilium/templates/cilium-agent/service.yaml
Comment (cyclinder) [reply install/kubernetes/cilium/templates/cilium-agent/service.yaml]: Nope. I think using `app.kubernetes.io/part-of` to uniformly declare which resources are part of cilium. And Add a new label(`app.kubernetes.io/name`) to replace `k8s-app` in the future (or co-exist with it) , they are both used for `selector` and `matchLabels`.

Comment (gandro) [reply install/kubernetes/cilium/templates/cilium-agent/service.yaml]: Your proposal of using `app.kubernetes.io/part-of` and `app.kubernetes.io/name` in the selectors sounds good to me!

**Comment by gandro** [comment]
> @gandro and @cyclinder I do agree. We do not want to deprecate `k8s-app` until we can safely do so, especially with its usage in `cilium sysdump`. That being said, we can add the recommended labels like mentioned above. There is a bit of value add here. Right now I have a kind cluster and I see that the 'agent' pods use `k8s-app=cilium` while the operators do not.
> 
> Ideally, we can do something like:
> 
> ```
> app.kubernetes.io/part-of: cilium
> app.kubernetes.io/name: cilium-agent
> ```
> 
> ```
> app.kubernetes.io/part-of: cilium
> app.kubernetes.io/name: cilium-operator
> ```
> 
> Tho, I'm not sure if s/name/component or not?

I like this proposal. I'm happy with `name`. 

@cyclinder Would you be willing to change this PR to reflect that?

**Comment by tommyp1ckles** [comment]
Changes lgtm, I am still a bit concerned over changing the immutable matchLabels. For example, if I try to do a helm upgrade with this branch I get:

```
Error: UPGRADE FAILED: cannot patch "cilium" with kind DaemonSet: DaemonSet.apps "cilium" is invalid: [spec.template.metadata.labels: Invalid value: map[string]string{"app.kubernetes.io/name":"cilium", "app.kubernetes.io/part-of":"cilium-agent", "k8s-app":"cilium"}: `selector` does not match template `labels`, spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/name":"cilium-agent", "k8s-app":"cilium"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable] && cannot patch "cilium-operator" with kind Deployment: Deployment.apps "cilium-operator" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/name":"cilium-operator", "io.cilium/app":"operator"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

**Review by ldelossa — APPROVED** [review]
This looks good to me, contingent on dealing with the existing feedback. Thanks for adding to the scope of this PR.

**Comment by cyclinder** [comment]
I'm sorry to not responding for a long time since I was busy with work these days.

> 
> @cyclinder Would you be willing to change this PR to reflect that?

Hi @gandro , If my understanding is right,  the current PR already reflects these changes, Right?

**Comment by cyclinder** [comment]
> Changes lgtm, I am still a bit concerned over changing the immutable matchLabels. For example, if I try to do a helm upgrade with this branch I get:
> 
> `Error: UPGRADE FAILED: cannot patch "cilium" with kind DaemonSet: DaemonSet.apps "cilium" is invalid: [spec.template.metadata.labels: Invalid value: map[string]string{"app.kubernetes.io/name":"cilium", "app.kubernetes.io/part-of":"cilium-agent", "k8s-app":"cilium"}: `selector`does not match template`labels`, spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/name":"cilium-agent", "k8s-app":"cilium"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable] && cannot patch "cilium-operator" with kind Deployment: Deployment.apps "cilium-operator" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/name":"cilium-operator", "io.cilium/app":"operator"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable`
> 
> Not sure what the best solution for this is, perhaps to have way to specify custom matchLabel selectors in values.yaml.

Thanks! Good catch! 

I'll test it on my local machine, and find a best solution for this.

**Comment by cyclinder** [comment]
Update:

As before, I added two labels to each cilium resource, `app.kubernetes.io/name` and `app.kubernetes.io/part-of`, and only when `UseNewMatchLabel`= `true`, then `app.kubernetes.io/name`  is one of the matchLabel.

We set the default value of `UseNewMatchLabel` to true in values.yaml , and we expect that when cilium is installed,  `app.kubernetes.io/name`will be one of the matchLabel.  We may failed when updating chart because matchLabel is immutable, so we need to update it by `--set UseNewMatchLabel=false`.

File: install/kubernetes/cilium/templates/hubble/peer-service.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/hubble/peer-service.yaml]: This does not look correct. This is intended to select the cilium DaemonSet pods, but the cilium pod does not have the `hubble-peer` label set.

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (gandro) [inline install/kubernetes/cilium/values.yaml.tmpl]: But overall I'm a bit nervous about adding this flag. Once it's enabled, it can never be disabled again, which is especially annoying because it prevents rollbacks to older versions of Cilium.

Maybe good to get a second opinion on this. @christarazi @aanm, thoughts?

---

Also, while this can default to true on new installations, we want to make sure we automatically set it to `false` for upgrading users. To do this, we have the `upgradeCompatibility` mechanism. You will have to add the same variable to all deployments which use this value, and ensure the label selectors are not changed if `upgradeCompatibility` is set to v1.11 or lower.

Check out an example here (note, the `$azureUsePrimaryAddress` is flipped around. For you, `$useNewMatchLabel` needs to default to false, and be set to true if `upgradeCompatibility` is set to v1.12 or higher).
https://github.com/cilium/cilium/blob/69be90059f9babb1757db862d6e0f25e3bf007ce/install/kubernetes/cilium/templates/cilium-configmap.yaml#L43-L48


You will also have to mention this new value in the upgrade notes:
https://github.com/cilium/cilium/blob/a0e0917c23dcf3e4fa4709b88f7cfb0e8c5ae6b0/Documentation/operations/upgrade.rst#helm-options

---

In addition, please rename this to `useNewMatchLabel` (lower case). 

I have also a few suggetions around the language here:
```suggestion
# -- Use the new label "app.kubernetes.io/name" as one of the selector labels.
# This adds "app.kubernetes.io/name" as one of the matchLabels when installing Cilium.
# It defaults to true on new installations. Because the matchLabels field is immutable,
# this must be set to false if upgrading from an installation of Cilium which did use the new
# matchLabels.
useNewMatchLabel: true
```

File: install/kubernetes/cilium/templates/hubble/metrics-service.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/hubble/metrics-service.yaml]: This does not look correct. This is intended to select the cilium DaemonSet pods, but the cilium pod does not have the `hubble` label set.

**Review by gandro — CHANGES_REQUESTED** [review]

**Comment by gandro** [comment]
> Hi @gandro , If my understanding is right, the current PR already reflects these changes, Right?

Yes, I think I missed that. The labels used in the current version look good the me.

File: install/kubernetes/cilium/templates/hubble/peer-service.yaml
Comment (cyclinder) [reply install/kubernetes/cilium/templates/hubble/peer-service.yaml]: Thanks!

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (cyclinder) [reply install/kubernetes/cilium/values.yaml.tmpl]: Thank you so much @gandro !  But I have a question about `upgradeCompatibility `. If `upgradeCompatibility ` is set to 1.13,  `$useNewMatchLabel ` will be true,  which means  if I upgrade from a lower version (e.g. 1.10) to 1.13, the upgrade will failed because of the new matchLabels used.

So I think the new matchLabel is enabled only if (`values.useNewMatchLabel` && `$useNewMatchLabel`) == `true`, Otherwise, It will not use new matchLabel. right?

Comment (gandro) [reply install/kubernetes/cilium/values.yaml.tmpl]: > So I think the new matchLabel is enabled only if (values.useNewMatchLabel && $useNewMatchLabel) == true, Otherwise, It will not use new matchLabel. right?

Yes, that would be the overall approach. However, given that it's not yet clear how users can actually upgrade to the new labels (what would be the required steps?), I'm tempted to postpone the matchLabel change. Would it be possible to only update the labels? We can then do the matchLabel change in a separate PR.

Comment (cyclinder) [reply install/kubernetes/cilium/values.yaml.tmpl]: Totally agree, I agree that now is not the time to add a new matchLabel. I will only update labels  in this PR. Thanks!

Comment (cyclinder) [reply install/kubernetes/cilium/values.yaml.tmpl]: Hi @gandro , Would you set some labels for this PR?  Such as  `/kind feature` etc. I found that I set it but it didn't work. :)

File: install/kubernetes/cilium/templates/cilium-operator/deployment.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-operator/deployment.yaml:10]: A quick grep for `name=operator` in the Cilium tree revealed that this label is used for the well-known identity feature. Therefore, it must not be removed. 

https://github.com/cilium/cilium/blob/19bda871b0fdaca7014be2988c45f4e2972bea02/pkg/identity/numericidentity.go#L285-L297

To be honest, I'm also not sure if adding additional labels breaks the well-known identities selector. That's something we also need to double check.

File: install/kubernetes/cilium/templates/etcd-operator/cilium-etcd-operator-deployment.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/etcd-operator/cilium-etcd-operator-deployment.yaml:7]: This is also used for assigning well-known identities to the etcd operator and therefore must not be removed.

**Review by gandro — CHANGES_REQUESTED** [review]
Thanks! There are still some changes needed

**Comment by gandro** [comment]
So I confirmed: Adding labels to the following components will break the well-known identities feature: https://docs.cilium.io/en/v1.11/concepts/terminology/#well-known-identities

If we add these new labels to `cilium-operator`, `etcd-operator`, `cilium-etcd`, we also have to update the label set used to determine those identities. In other words, we will have to add a third label selector with the new labels to each of the affected identities here (similar to how it was done in https://github.com/cilium/cilium/pull/16585):

https://github.com/cilium/cilium/blob/19bda871b0fdaca7014be2988c45f4e2972bea02/pkg/identity/numericidentity.go#L183

**Comment by cyclinder** [comment]
> 

Thanks for your patient reply！I checked this part of the source code, and as you said, there are four places should to update the identities : `cilium-operator`, `etcd-operator`, `cilium-etcd`, and `cilium-etcd-operator`.

But only `cilium-operator` and `cilium-etcd-operator `correspond to the above in PR, it seems that `etcd-operator`, and `cilium-etcd`, we do not need to update?

**Comment by cyclinder** [comment]
> So I confirmed: _Adding any new label_ to the following components will break the well-known identities feature: https://docs.cilium.io/en/v1.9/concepts/terminology/#well-known-identities
> 
> If we add these new labels to `cilium-operator`, `etcd-operator`, `cilium-etcd`, and `cilium-etcd-operator`, we will also have to update the label set used to determine those identities. In other words, we will have to add a third label selector with the new labels to each of the affected identities here (similar to how it was done in #16585):
> 
> https://github.com/cilium/cilium/blob/19bda871b0fdaca7014be2988c45f4e2972bea02/pkg/identity/numericidentity.go#L183

Thanks for your patience! I checked this part of the source code, and as you said, there are `cilium-operator`, `etcd-operator`, `cilium-etcd`, and `cilium-etcd-operator` where we need to update the identities.

But only `cilium-operator` and `cilium-etcd-operator` correspond to the above in PR, it seems that `etcd-operator` and `cilium-etcd`, we do not need to update?

**Comment by gandro** [comment]
> But only `cilium-operator` and `cilium-etcd-operator` correspond to the above in PR, it seems that `etcd-operator` and `cilium-etcd`, we do not need to update?

Yes, I think you are correct. The other ones are deprecated (their deployment manifest is [here](https://github.com/cilium/cilium-etcd-operator/)) and therefore not changed by this PR.

File: pkg/identity/numericidentity.go
Comment (gandro) [inline pkg/identity/numericidentity.go:297]: This unfortunately does not work. We want to keep the existing label selector as is, and create a new one for the extended labels.

You will have change the code in such a way that it supports both the old and the new label set.  Unfortunately, I just realized that we have to support both the versions with and without the `PodNamespaceMetaNameLabel`, because I believe we want to support K8s older than v1.21 still.

So in the end, there will have to be 4 calls to `WellKnown.add`, one for each label set:

```
WellKnown.add(ReservedCiliumOperator, ciliumOperatorLabels)
WellKnown.add(ReservedCiliumOperator2, ciliumOperatorLabelsWithNamespace)
WellKnown.add(ReservedCiliumOperator3, ciliumOperatorLabelsWithRecommendedLabels)
WellKnown.add(ReservedCiliumOperator4, ciliumOperatorLabelsWithRecommendedLabelsAndNamespace)
```

However, this will double the number of reserved identities related to the operator and etcd server. I'm not sure this is an approach that scale. Thoughts @aanm?

Comment (aanm) [reply pkg/identity/numericidentity.go:297]: That's a good question, but then I wonder: The label changes will part of Cilium version X.Y.Z and the well known identities will also be part of the version X.Y.Z, so why would we need to duplicate them?

Comment (gandro) [reply pkg/identity/numericidentity.go:297]: I guess there could be a time during the rollout where some agent pods still use the old well-known identity definition, while the operator or etcd server has already been upgraded to the new labels. Especially for etcd, that might cause some disruption during the upgrade, because none of the old agents will be able to connect to the new etcd server until they have been upgraded too.

Comment (aanm) [reply pkg/identity/numericidentity.go:297]: It something that we would need to double check but I would guess that wouldn't be a problem because the security identity would end up being the same.

Comment (gandro) [reply pkg/identity/numericidentity.go:297]: So, if we're not worried about supporting the label set of versions N+1 or N-1, then this change here is good as is.

File: install/kubernetes/cilium/templates/etcd-operator/poddisruptionbudget.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/etcd-operator/poddisruptionbudget.yaml:10]: For consistency, let's also keep the labels on the poddisruptionbudget here. I think we could remove them from all components going forward (as long as we also update the well-known security identities), but I'd like to do that in a separate PR, as I think that might be a bigger undertaking (I also see those labels being used e.g. in CI).

File: install/kubernetes/cilium/templates/cilium-operator/service.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-operator/service.yaml:9]: For consistency, let's also keep the labels on the service here.

File: install/kubernetes/cilium/templates/cilium-operator/poddisruptionbudget.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-operator/poddisruptionbudget.yaml:10]: For consistency, let's also keep the labels on the poddisruptionbudget here.

File: install/kubernetes/cilium/templates/etcd-operator/poddisruptionbudget.yaml
Comment (cyclinder) [reply install/kubernetes/cilium/templates/etcd-operator/poddisruptionbudget.yaml:10]: @gandro Thanks!!  I miss this.

**Review by tommyp1ckles — APPROVED** [review]
LGTM, thanks you!

File: pkg/identity/numericidentity.go
Comment (gandro) [inline pkg/identity/numericidentity.go]: Given our findings in https://github.com/cilium/cilium/pull/20213#discussion_r915603889, this is actually not needed. My comment about `this change here is good as is` was referring to the previous version.

All you need to change is the `ciliumOperatorLabels` to add the two labels to both `ReservedCiliumOperator` and `ReservedCiliumOperator2`. `ReservedCiliumOperator3` and `ReservedCiliumOperator4` are not needed.

**Review by gandro — CHANGES_REQUESTED** [review]
Thanks. Given the discussion, the well-known identity change can be simplified again. Let's not add the additional identities.

**Review by gandro — APPROVED** [review]
Thanks. This should be good to go form my side I think. Let's run CI again.

**Comment by gandro** [comment]
@cyclinder It seems like your `signed-off-by` line in the commit is not properly formatted. From the `checkpatch` CI ([here](https://github.com/cilium/cilium/runs/7285107528?check_suite_focus=true)):

```
  Error: ERROR:BAD_SIGN_OFF: Unrecognized email address: 'cyclinder qifeng.guo@daocloud.io'
  #7: 
  Signed-off-by: cyclinder qifeng.guo@daocloud.io
  
  Error: ERROR:NO_AUTHOR_SIGN_OFF: Missing Signed-off-by: line by nominal patch author 'cyclinder <qifeng.guo@daocloud.io>'
```
It's missing the `<` and `>` around your email address. You can add this via git CLI by using `git commit --amend -s`

**Comment by cyclinder** [comment]
@gandro Thank you very much for your help!

**Comment by cyclinder** [comment]
I have no idea about this CI failed([gke-stable](https://jenkins.cilium.io/job/Cilium-PR-K8s-GKE/8910/)). Can this CI job be retested?

**Comment by cyclinder** [comment]
Thanks, Does this PR require a review?  This is my first contribution to cilium :) , Can you tell me about the requirements for PR merged?

**Comment by gandro** [comment]
I prefer to have another code review from the policy CODEOWNERs (e.g. Chris), to ensure the well-known identity change is valid.

**Review by christarazi — APPROVED** [review]
LGTM for well-known identities. Given that the security identity value will not change (just the labels that map to the identity value), then this makes it safe for upgrading / downgrading. I've added the `upgrade-impact` label just in case we need to review any PRs in the future regarding any changes on upgrade.

**Pull request merged**
